.vsts-pipelines/builds/ci.yml

```diff
@@ -3,41 +3,14 @@ trigger:
 - release/*
 
 phases:
-- phase: Windows
-  queue:
-    name: Hosted VS2017
-    demands: cmd
-  steps:
-  - script: build.cmd -ci
-  - task: PublishTestResults@2
-    condition: always()
-    inputs:
-      testRunner: vstest
-      testResultsFiles: 'artifacts/logs/**/*.trx'
+- template: ../templates/phases/default-build.yml
+  parameters:
+    agentOs: macOS
 
+- template: ../templates/phases/default-build.yml
+  parameters:
+    agentOs: Linux
 
-- phase: macOS
-  queue:
-    name: Hosted macOS Preview
-    demands: agent.os -equals Darwin
-  steps:
-  - script: ./build.sh --ci
-  - task: PublishTestResults@2
-    condition: always()
-    inputs:
-      testRunner: vstest
-      testResultsFiles: 'artifacts/logs/**/*.trx'
-
-
-- phase: Linux
-  queue:
-    name: Hosted Linux Preview
-    demands: agent.os -equals Linux
-  steps:
-  - script: ./build.sh --ci
-  - task: PublishTestResults@2
-    condition: always()
-    inputs:
-      testRunner: vstest
-      testResultsFiles: 'artifacts/logs/**/*.trx'
-
+- template: ../templates/phases/default-build.yml
+  parameters:
+    agentOs: Windows
```

Comment:
Is there any particular reason for keeping the `_NT` part? Or is this just convention?

Reply:
I tend to align with the values that our VSTS agent pools provide via demands, so it's consistent between the demands I specify for my queue, and my parameters.

Windows -> Windows_NT
Linux -> Linux
Mac -> Darwin
 
Currently, for the hosted pools, the machine types are separated by OS, so that doesn't specifically apply and I don't even think the Agent.Os is listed in the dotnet.visualstudio.com hosted machine pools.  I'm not certain where the demand values come from or if they were just arbitrarily defined.

Historically, the demands were used to determine what machine OS you want and all OS legs are pointing to the same agent pool.

https://github.com/dotnet/arcade/blob/master/.vsts-dotnet.yml#L8-L9

Reply:
Ok, that sounds like sound reasoning. So, should agentOs be `{ Windows_NT | Linux | Darwin }`?

by the way, you may be interested in this: https://developercommunity.visualstudio.com/content/problem/251502/the-agentos-capability-is-missing-on-hosted-or-hos.html